sgkit/io/vcf/vcf_reader.py

```diff
@@ -105,8 +107,14 @@ def vcf_to_zarr_sequential(
                     max_variant_allele_length, max(len(x) for x in alleles)
                 )
 
-                gt = variant.genotype.array()
-                call_genotype[i] = gt[..., 0:-1]
+                fill = -2 if mixed_ploidy else -1
+                gt = variant.genotype.array(fill=fill)
+                gt_length = gt.shape[-1] - 1
```

Comment:
I wasn't sure why 1 was being subtracted here at first. Maybe add a comment saying the last entry indicates if the genotype is phased. (I should have done so in the original code!)